crates/storage-api/src/invocation_status_table/mod.rs

```diff
@@ -335,6 +444,9 @@ pub struct CompletedInvocation {
     pub idempotency_key: Option<ByteString>,
     pub timestamps: StatusTimestamps,
     pub response_result: ResponseResult,
+
+    /// Used by the Table implementation to pick where to write
+    pub source_table: SourceTable,
```

Comment:
> The field needs to be public because we have the separation of storage-api and partition-store crate, right?

Not only, that's because I also need to read it in the state machine.

> Seems like we are repeating the same information for every inner enum variant type.

Yes. I didn't wanted to change the overall structure of `InvocationStatus` (that's a non small refactor, so I avoided it for now)